state/stateenvirons/policy.go

```diff
@@ -75,28 +105,16 @@ func (p environStatePolicy) ProviderConfigSchemaSource(cloudName string) (config
 }
 
 // ConstraintsValidator implements state.Policy.
-func (p environStatePolicy) ConstraintsValidator(ctx context.ProviderCallContext) (constraints.Validator, error) {
-	model, err := p.st.Model()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	if model.Type() == state.ModelTypeIAAS {
-		env, err := p.getEnviron(model)
-		if err != nil {
-			return nil, err
-		}
-		return env.ConstraintsValidator(ctx)
-	}
-	broker, err := p.getBroker(model)
+func (p *environStatePolicy) ConstraintsValidator(ctx context.ProviderCallContext) (constraints.Validator, error) {
+	checker, err := p.getChecker()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return broker.ConstraintsValidator(ctx)
+	return checker.ConstraintsValidator(ctx)
 }
 
 // InstanceDistributor implements state.Policy.
-func (p environStatePolicy) InstanceDistributor() (context.Distributor, error) {
+func (p *environStatePolicy) InstanceDistributor() (context.Distributor, error) {
```

Comment:
This doesn't make any calls to get instance types so in theory doesn't benefit from using the cached environ.
Add a comment saying that it has been deliberately left out.